### PR TITLE
MISUV-2940 Make .get throw an exception if data doesn't exist in Tax year template files 

### PR DESCRIPTION
**Pull request by andriikondiuk**

### CODE REVIEW

**Review by chandraAndem — CHANGES_REQUESTED** [review]
Couple of changes needed.

File: app/views/TaxYearOverview.scala.html [hunk 1]
```diff
@@ -72,48 +73,58 @@
...
-                        <td class="govuk-table__cell">@documentDetailWithDueDate.dueDate.get.toLongDateShort</td>
-                        <td class="govuk-table__cell">@messages(s"tax-year-overview.payments.${documentDetailWithDueDate.documentDetail.getChargePaidStatus}")</td>
-                        <td class="govuk-table__cell">@documentDetailWithDueDate.documentDetail.originalAmount.get.toCurrencyString</td>
-                    }
-                </tr>
-            }
-            </tbody>
-        </table>
-    }
+                        @documentDetailWithDueDate.documentDetail.amountCodedOut.map { amountCodedOut =>
```
[the remark this answers is not in the document]
Comment (andriikondiuk) [reply app/views/TaxYearOverview.scala.html]: Done. thanks

File: app/views/TaxYearOverview.scala.html [hunk 2]
```diff
@@ -72,48 +73,58 @@
...
+                        <td class="govuk-table__cell">@amountCodedOut.toCurrencyString</td>
+                        }
+                } else {
+                        @documentDetailWithDueDate.dueDate.map { dueDate =>
```
[the remark this answers is not in the document]
Comment (andriikondiuk) [reply app/views/TaxYearOverview.scala.html]: Done. thanks

File: app/views/TaxYearOverview.scala.html [hunk 3]
```diff
@@ -73,48 +73,55 @@
...
+                    } else {
+                        <a class="govuk-link"  id="paymentTypeLink-@{index}" href="@getChargeSummaryHref(documentDetailWithDueDate)">@messages(s"tax-year-overview.payments.${documentDetailWithDueDate.documentDetail.getChargeTypeKey(codingOutEnabled)}")</a>
+                    }
+                </div>
+                    @if(documentDetailWithDueDate.dunningLock){<div class="form-hint govuk-body-s"> @messages("tax-year-overview.payments.paymentUnderReview") </div>}
+                    @if(documentDetailWithDueDate.documentDetail.hasLpiWithDunningBlock){<div class="form-hint govuk-body-s"> @messages("tax-year-overview.payments.paymentUnderReview") </div>}
+                </td>
+                @if(documentDetailWithDueDate.isLatePaymentInterest) {
+                    @documentDetailWithDueDate.documentDetail.latePaymentInterestAmount.map { latePaymentInterestAmount =>
+                        @documentDetailWithDueDate.dueDate.map { dueDate =>
```
Comment (chandraAndem) [inline app/views/TaxYearOverview.scala.html:102]: Sorry missed this one in first review. This dueDate needs to throw an exception as well.

File: app/views/TaxYearOverview.scala.html [hunk 4]
```diff
@@ -73,48 +73,55 @@
...
+                        <td class="govuk-table__cell">@documentDetailWithDueDate.dueDate.getOrElse(throw MissingFieldException("Due Date")).toLongDateShort</td>
                         <td class="govuk-table__cell">@messages(s"tax-year-overview.payments.${documentDetailWithDueDate.documentDetail.getChargePaidStatus}")</td>
-                        <td class="govuk-table__cell">@documentDetailWithDueDate.documentDetail.originalAmount.getOrElse(throw new MissingFieldException("Original Amount")).toCurrencyString</td>
-                    }
-                </tr>
-            }
-            </tbody>
-        </table>
-    }
+                        @documentDetailWithDueDate.documentDetail.originalAmount.map { originalAmount =>
```
Comment (chandraAndem) [inline app/views/TaxYearOverview.scala.html:115]: Not sure why its changed from last review but this one needs throwing exception as well.

**Review by chandraAndem — CHANGES_REQUESTED** [review]
Couple of changes required.

File: app/views/TaxYearOverview.scala.html (hunk 4 above)
Comment (andriikondiuk) [reply app/views/TaxYearOverview.scala.html]: All should be done now. Thanks

**Review by chandraAndem — APPROVED** [review]

**Pull request merged**